pex/third_party/__init__.py

```diff
@@ -116,10 +119,11 @@ def iter_root_packages(self, relpath):
       yield package
 
   def _filter_names(self, relpath, pattern, group):
-    pat = re.compile(r'^{prefix}{pattern}$'
-                     .format(prefix=self.prefix + ((relpath + os.sep) if relpath else ''),
-                             pattern=pattern))
-
+    # We replace os.sep with '/' here because the zip file format spec specifies that paths must
+    # use forward slashes. See section 4.4.17 of
+    # https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT.
+    relpath_pat = '' if not relpath else '{}{}'.format(relpath, os.sep).replace(os.sep, '/')
```

Comment:
Nit, but, like above, why use os.sep when you don't want it?
```python
relpath_pat = '' if not relpath else '{}/'.format(relpath.replace(os.sep, '/'))
```

Reply:
Fixed!